Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1310,23 +1304,34 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, unsigned int dstFo
     copy_to_vram = false;
   }
 
-  // Invalidate all textures that overlap the range of our efb copy.
-  // Unless our efb copy has a weird stride, then we mark them to check for partial texture updates.
-  // TODO: This also invalidates partial overlaps, which we currently don't have a better way
-  //       of dealing with.
-  bool invalidate_textures = dstStride == bytes_per_row || !copy_to_vram;
+  // Invalidate all textures, if they are either fully overwritten by our efb copy, or if they
+  // have a different stride than our efb copy. Textures with the same stride and that are partly
+  // overwritten are marked to check them for partial texture updates.
+  // TODO: The logic to detect a full overlap of strided efb copies is not 100% accurate
+  bool strided_efb_copy = dstStride != bytes_per_row;
   auto iter = FindOverlappingTextures(dstAddr, covered_range);
   while (iter.first != iter.second)
   {
     TCacheEntry* entry = iter.first->second;
     if (entry->OverlapsMemoryRange(dstAddr, covered_range))
     {
-      if (invalidate_textures)
+      u32 overlap_range = std::min(entry->addr + entry->size_in_bytes, dstAddr + covered_range) -
+                          std::max(entry->addr, dstAddr);
+      if (entry->memory_stride != dstStride || !copy_to_vram ||
+          (!strided_efb_copy && entry->size_in_bytes == overlap_range) ||
```

Comment:
Oh, i want to check 2 different stride related things here:
1. Check if the stride of the efb copy and the overlapping texture are compatible
2. Try to track if the efb copy and the overlapping texture are overlapping 100%. Those strided efb copies could do weird things, since they don't actually overwrite all data from their address to address + size, which is why it's called covered_range instead of size here. (but not in the entry's properties...) If you got an idea on how to properly track if the efb copy overwrites the other texture to 100%, please help me here.